File: php/features.php
<?php
declare(strict_types=1);

// MysqlVisualExplain SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MysqlVisualExplainFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MysqlVisualExplainBaseFeature();
            case "test":
                return new MysqlVisualExplainTestFeature();
            default:
                return new MysqlVisualExplainBaseFeature();
        }
    }
}

File: py/features.py
# MysqlVisualExplain SDK feature factory

from feature.base_feature import MysqlVisualExplainBaseFeature
from feature.test_feature import MysqlVisualExplainTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MysqlVisualExplainBaseFeature(),
        "test": lambda: MysqlVisualExplainTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
